### DISCUSSION TITLE
My api crashes when I send a request without a file

### DISCUSSION THREAD

**Discussion by rvyh** [vercel/next.js#24203] [Help] [answered]
I have a form with some text inputs and an optional file input. My api crashes when I send the form without a file. (When I send the file too, it works fine!)

Sending the form generates a POST request to an endpoint with the following code:

```
import nextConnect from 'next-connect';
var multer  = require('multer');

// initialize api-route handler
const handler = nextConnect();

var storage = multer.diskStorage({
  destination: function (req, file, cb) {
    cb(null, 'uploads/');
  },
  filename: function (req, file, cb) {
    cb(null, file.originalname);
  }
})

// configure multer for file upload
// store uploaded image on uploads/
const upload = multer({ 
  //dest: 'uploads/',
  limits: {
    fields: 6,
    files: 1,
    fileSize: 52224
  },
  storage
})

// use multer middleware
// The single file will be stored in req.file.
handler.use(upload.single('logo'))

handler.post((req, res) => {
  // todo
  res.json({ result: "success" });
})

export const config = {
  api: {
      bodyParser: false,
  },
}

export default handler;
```

When not uploading a file, I tried replacing this

`handler.use(upload.single('logo'))`

with

`handler.use(upload.none());`

but I got the same error:

POST http://localhost:3000/api/endpoint 500 (Internal Server Error)
Uncaught (in promise) SyntaxError: Unexpected token T in JSON at position 0

**Comment by akellbl4** [comment]
Hi,
You can try to catch error if it occurs by multer like this:
```js
 multer({
    onError(err, next) {
      console.log('error', err);
      next(err);
    }
  })
  ```
  And another way to catch error on `next-connect` level:
  ```js
const handler = nextConnect({
    onError(err, req, res, next) {
        console.log('error', err);
        next(err);
    }
});

**Reply by rvyh** [reply]
Now I have:

```
function onError(err, req, res, next) {
  console.log(err);
  res.status(500).end(err.toString());
}

const handler = nextConnect({ onError });
```

And I get

`MulterError: Too many fields`

**Comment by rvyh** [comment, marked as answer]
I found that increasing the number of fields solves the "**MulterError: Too many fields**" error.

```
limits: {
    fields: 7,
```

Although [multer documentation](https://www.npmjs.com/package/multer) says 

fields = Max number of non-file fields

I have to enter the total number of fields, including file fields.

But the following issue is still remaining. I have to dynamically run

handler.use(upload.single('logo'))

or

handler.use(upload.none());

depending on whether the user submits the form with or without a file, respectively.

How to do this?

**Reply by akellbl4** [reply]
Looks like you add not the only files to the `FormData` while uploading files. Check how you are sending data on the frontend.

**Reply by rvyh** [reply]
Yes, I had something wrong in the frontend. I don't get any errors now, thanks for the help!

**Reply by akellbl4** [reply]
Please, mark the discussion as resolved if you solved your problems ;)
